docs/known_issues.md

```diff
@@ -14,15 +14,24 @@ As a workaround, use `tag.turbo_frame` instead of `turbo_frame_tag`.
 
 Note: For the same functionality as `turbo_frame_tag(my_model)`, use `tag.turbo_frame(id: dom_id(my_model))`.
 
-## form_for compatibility
+## Compatibility with Rails form helpers
 
 ViewComponent [isn't compatible](https://github.com/viewcomponent/view_component/issues/241) with `form_for` helpers by default.
 
-## Inconsistent controller rendering behavior between Rails versions
+Passing a form object reference (usually `f`) to a ViewComponent works for simple cases, like `f.text_field :name`.
+Content will be ill-ordered or duplicated in complex cases (such as passing blocks to form helpers or when nesting components).
 
-In versions of Rails < 6.1, rendering a ViewComponent from a controller doesn't include the layout.
+Consider following options:
 
-## Forms don't use the default form builder
+- Render a form end-to-end within one ViewComponent
+- Render a [classical view partial](https://guides.rubyonrails.org/layouts_and_rendering.html#using-partials) within the ViewComponent, which includes the form
+- Use a [custom `FormBuilder`](https://guides.rubyonrails.org/form_helpers.html#customizing-form-builders) to create reusable form components.
+  Note: There are multiple approaches how to implement a FormBuilder:
+  - Without FormBuilder at all, where one would [use Action View helpers](https://api.rubyonrails.org/classes/ActionView/Helpers/FormBuilder.html) to create the desired output
+  - Using a FormBuilder that overrides all field helpers to render a ViewComponent, so that you can customize each field individually (for example, [view_component-form gem](https://github.com/pantographe/view_component-form))
+  - Using a lightweight implementation of ViewComponent. For example, [Primer ViewComponents](https://github.com/primer/view_components) implemented with [`ActsAsComponent`](https://github.com/primer/view_components/blob/main/lib/primer/forms/acts_as_component.rb) a lightweight version, which they use in the context of `FormBuilder`.
```

Comment:
```suggestion
- Render an entire form within a single ViewComponent.
- Render a [partial](https://guides.rubyonrails.org/layouts_and_rendering.html#using-partials) within the ViewComponent which includes the form.
- Use a [custom `FormBuilder`](https://guides.rubyonrails.org/form_helpers.html#customizing-form-builders) to create reusable form components:
  - Without FormBuilder at all, [use Action View helpers](https://api.rubyonrails.org/classes/ActionView/Helpers/FormBuilder.html) to create the desired output
  - Using a FormBuilder that overrides all field helpers to render a ViewComponent, so that each field can be customized individually (for example, [view_component-form](https://github.com/pantographe/view_component-form))
  - Using a lightweight implementation of ViewComponent. For example, [Primer ViewComponents](https://github.com/primer/view_components) implemented [`ActsAsComponent`](https://github.com/primer/view_components/blob/main/lib/primer/forms/acts_as_component.rb) which is used in the context of `FormBuilder`.
```

Reply:
> Render a [partial](https://guides.rubyonrails.org/layouts_and_rendering.html#using-partials) which includes the form, within the ViewComponent.

What about reordering this? Because it's the partial that includes the form, not the ViewComponent.

Reply:
> Without FormBuilder at all

It feels weird that this option is introduced as a sub-item of "Use a custom `FormBuilder`"

Reply:
> Without FormBuilder at all

This statement is just wrong. I need to fix it.